src/main/java/org/generationcp/middleware/service/impl/study/StudyInstanceServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package org.generationcp.middleware.service.impl.study;
+
+import org.generationcp.middleware.domain.dms.ExperimentType;
+import org.generationcp.middleware.domain.dms.ExperimentValues;
+import org.generationcp.middleware.domain.etl.MeasurementVariable;
+import org.generationcp.middleware.domain.oms.TermId;
+import org.generationcp.middleware.domain.ontology.VariableType;
+import org.generationcp.middleware.enumeration.DatasetTypeEnum;
+import org.generationcp.middleware.hibernate.HibernateSessionProvider;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.generationcp.middleware.manager.Operation;
+import org.generationcp.middleware.operation.saver.ExperimentModelSaver;
+import org.generationcp.middleware.pojos.Location;
+import org.generationcp.middleware.pojos.dms.ExperimentModel;
+import org.generationcp.middleware.pojos.dms.Geolocation;
+import org.generationcp.middleware.pojos.dms.GeolocationProperty;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.generationcp.middleware.service.api.study.StudyInstanceService;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Transactional
+public class StudyInstanceServiceImpl implements StudyInstanceService {
+
+	public StudyInstanceServiceImpl() {
+		// no-arg constuctor is required by CGLIB proxying used by Spring 3x and older.
+	}
+
+	public StudyInstanceServiceImpl(final HibernateSessionProvider sessionProvider) {
+		this.sessionProvider = sessionProvider;
+		this.daoFactory = new DaoFactory(sessionProvider);
+		this.experimentModelSaver = new ExperimentModelSaver(sessionProvider);
+	}
+
+	private HibernateSessionProvider sessionProvider;
```

Comment:
We should not use **ExperimentModelSaver**  anymore and any xxSaver, xxBuilder, xxDestroyer etc.. classes).
See if you can use or overload **ExperimentModelGenerator#generate** as we do in experimental design saving

Reply:
Done.